python-lib/sharepoint_client.py

```diff
@@ -113,10 +111,19 @@ def setup_login_details(self, login_details):
         logger.info("SharePointClient:sharepoint_root={}".format(self.sharepoint_root))
 
     def setup_sharepoint_online_url(self, login_details):
-        self.sharepoint_tenant = login_details['sharepoint_tenant']
-        logger.info("SharePointClient:sharepoint_tenant={}".format(self.sharepoint_tenant))
-        self.sharepoint_url = self.sharepoint_tenant + ".sharepoint.com"
-        self.sharepoint_origin = "https://" + self.sharepoint_url
+        scheme, domain, tenant = parse_url(login_details['sharepoint_tenant'])
+        if scheme:
+            self.sharepoint_url = domain
+            self.sharepoint_origin = scheme + "://" + domain
+        else:
+            self.sharepoint_url = tenant + ".sharepoint.com"
```

Comment:
It could be interesting to cover the case where the tenant in the preset is something like ikuiku.sharepoint.com (sharepoint.com or sharepoint is in the tenant but a custom domain prefix is also provided)